src/Reflection/ReflectionParameter.php

```diff
@@ -373,17 +373,15 @@ public function getPosition(): int
      *
      * (note: this has nothing to do with DocBlocks).
      */
-    public function getType(): ?ReflectionType
+    public function getType(): ReflectionNamedType|ReflectionUnionType|null
     {
         $type = $this->node->type;
 
         if ($type === null) {
             return null;
         }
 
-        $allowsNull = $this->isDefaultValueAvailable() && $this->getDefaultValue() === null && ! $this->isDefaultValueConstant();
```

Comment:
@WyriHaximus I think this is necessary. The test `testPhpTypeDeclarationWithNullDefaultValueAllowsNull` is failing because you removed this.

Reply:
It is, good catch, not sure why I removed it tbh 👍